argocd/datadog_checks/argocd/data/conf.yaml.example

```diff
@@ -0,0 +1,631 @@
+    ## which uses the scheme to decide whether to do the DNS resolution on
+    ## the client or proxy. If you want to resolve the domains on the proxy
+    ## server, use `socks5h` as the scheme.
+    #
+    # proxy:
+    #   http: http://<PROXY_SERVER_FOR_HTTP>:<PORT>
+    #   https: https://<PROXY_SERVER_FOR_HTTPS>:<PORT>
+    #   no_proxy:
+    #   - <HOSTNAME_1>
+    #   - <HOSTNAME_2>
+
+    ## @param skip_proxy - boolean - optional - default: false
+    ## If set to `true`, this makes the check bypass any proxy
+    ## settings enabled and attempt to reach services directly.
+    #
+    # skip_proxy: false
+
+    ## @param timeout - number - optional - default: 10
+    ## The timeout for connecting to services.
+    #
+    # timeout: 10
+
+    ## @param service - string - optional
+    ## Attach the tag `service:<SERVICE>` to every metric, event, and service check emitted by this integration.
+    ##
+    ## Additionally, this sets the default `service` for every log source.
+    #
+    # service: <SERVICE>
+
+## Every instance is scheduled independently of the others.
+#
+instances:
+
+  -
+    ## @param app_controller_endpoint - string - optional
+    ## Endpoint exposing the Application Controller's Prometheus metrics.
+    #
+    # app_controller_endpoint: http://argocd-metrics:8082/metrics
+
+    ## @param api_server_endpoint - string - optional
+    ## Endpoint exposing the API Server's Prometheus metrics.
+    #
+    # api_server_endpoint: http://argocd-server-metrics:8083/metrics
+
+    ## @param repo_server_endpoint - string - optional
+    ## Endpoint exposing the Repo Server's Prometheus metrics.
+    #
+    # repo_server_endpoint: http://argocd-repo-server:8084/metrics
+
+    ## @param raw_metric_prefix - string - optional
+    ## A prefix that will be removed from all exposed metric names, if present.
+    ## All configuration options will use the prefix-less name.
+    #
+    # raw_metric_prefix: <PREFIX>_
+
+    ## @param extra_metrics - (list of string or mapping) - optional
+    ## This list defines metrics to collect from the `openmetrics_endpoint`, in addition to
+    ## what the check collects by default. If the check already collects a metric, then
+    ## metric definitions here take precedence. Metrics may be defined in 3 ways:
+    ##
+    ## 1. If the item is a string, then it represents the exposed metric name, and
+    ##    the sent metric name will be identical. For example:
+    ##
+    ##      extra_metrics:
+    ##      - <METRIC_1>
+    ##      - <METRIC_2>
+    ## 2. If the item is a mapping, then the keys represent the exposed metric names.
+    ##
+    ##      a. If a value is a string, then it represents the sent metric name. For example:
+    ##
+    ##           extra_metrics:
+    ##           - <EXPOSED_METRIC_1>: <SENT_METRIC_1>
+    ##           - <EXPOSED_METRIC_2>: <SENT_METRIC_2>
+    ##      b. If a value is a mapping, then it must have a `name` and/or `type` key.
```

Comment:
@buraizu I'll have to take a closer look at these since these are imported from a template. I think I'll have to make changes to that template and regenerate these.

Reply:
@buraizu I started looking into this and will try to put up PR with the edit you had commented on. For this one I'm a bit unsure. Can you elaborate a bit more on what you would like to see or which part is a bit unclear? Since I have context on this, when I try reading it, I can't pick out what's unclear. Thanks in advance!

Reply:
@steveny91 thanks for following up -- I'm referencing the `and/or` part. Could you confirm if both the `name` and `type` key are required, or if it's possible to use only one without the other?